cni/network/network.go

```diff
@@ -1341,20 +1342,28 @@ func convertNnsToCniResult(
 
 			intIndex := i
 			for _, ip := range ni.Ipaddresses {
+				ipAddr := net.ParseIP(ip.Ip)
 
-				ipWithPrefix := fmt.Sprintf("%s/%s", ip.Ip, ip.PrefixLength)
-				_, ipNet, err := net.ParseCIDR(ipWithPrefix)
+				prefixLength, err := strconv.Atoi(ip.PrefixLength)
 				if err != nil {
-					log.Logger.Error("Error while converting to cni result",
+					log.Logger.Error("Error parsing prefix length while converting to cni result",
+						zap.String("prefixLength", ip.PrefixLength),
 						zap.String("operation", operationName),
 						zap.String("pod", podName),
 						zap.Error(err))
 					continue
 				}
 
+				var address net.IPNet
+				if ipAddr.To4() != nil {
+					address = net.IPNet{IP: ipAddr, Mask: net.CIDRMask(prefixLength, 32)}
+				} else {
+					address = net.IPNet{IP: ipAddr, Mask: net.CIDRMask(prefixLength, 128)}
+				}
```

Comment:
It's more idiomatic to eliminate `else` where possible in Go

```suggestion
				address := net.IPNet{
					IP: ipAddr,
					Mask: net.CIDRMask(prefixLength, 128)
				}
				
				if ipAddr.To4() != nil {
					address.Mask = net.CIDRMask(prefixLength, 32)
				}
```

Reply:
Updated pull request with suggested changes.